ios/Classes/DkNappDrawerDrawer.m

```diff
@@ -100,23 +104,59 @@ - (MMDrawerController *)controller
     __weak __typeof__(self) weakSelf = self;
 
     if (leftWindow != nil) {
+
+      //both left and right
       if (rightWindow != nil) {
-        //both left and right
+
+        TiViewController *leftController = ControllerForViewProxy(leftWindow);
+        TiViewController *rightController = ControllerForViewProxy(rightWindow);
+
+        TiUIiOSNavWindowProxy *centerProxy = [self.proxy valueForUndefinedKey:@"centerWindow"];
+
+        TiThreadPerformOnMainThread(^{
+          [centerProxy windowWillOpen];
+          [centerProxy windowDidOpen];
+        },
+            YES);
+
         controller = [[CustomMMDrawerController alloc] initWithCenterViewController:centerWindow
-                                                           leftDrawerViewController:ControllerForViewProxy(leftWindow)
-                                                          rightDrawerViewController:ControllerForViewProxy(rightWindow)];
-      } else {
+                                                           leftDrawerViewController:leftController
+                                                          rightDrawerViewController:rightController];
         //left only
+      } else {
+
+        TiViewController *leftController = ControllerForViewProxy(leftWindow);
+
+        TiUIiOSNavWindowProxy *centerProxy = [self.proxy valueForUndefinedKey:@"centerWindow"];
+
+        TiThreadPerformOnMainThread(^{
+          [centerProxy windowWillOpen];
+          [centerProxy windowDidOpen];
+        },
+            YES);
+
         controller = [[CustomMMDrawerController alloc] initWithCenterViewController:centerWindow
-                                                           leftDrawerViewController:ControllerForViewProxy(leftWindow)];
+                                                           leftDrawerViewController:leftController];
       }
-    } else if (rightWindow != nil) {
       //right only
+    } else if (rightWindow != nil) {
+
+      TiViewController *rightController = ControllerForViewProxy(rightWindow);
+
+      TiUIiOSNavWindowProxy *centerProxy = [self.proxy valueForUndefinedKey:@"centerWindow"];
+
+      TiThreadPerformOnMainThread(^{
+        [centerProxy windowWillOpen];
+        [centerProxy windowDidOpen];
+      },
+          YES);
+
       controller = [[CustomMMDrawerController alloc] initWithCenterViewController:centerWindow
-                                                        rightDrawerViewController:ControllerForViewProxy(rightWindow)];
-    } else {
+                                                        rightDrawerViewController:rightController];
+
       //error
-      NSLog(@"[ERROR] NappDrawer: No windows assigned");
+    } else {
+      NSLog(@"[DkNappDrawerDrawer][ERROR] No windows assigned");
```

Comment:
For proper error reporting on the console, the `[ERROR]` prefix should be first.